Add support for version pinning

actionjack opened the pull request:
Add support for version pinning of docker_registry

**What**

Add support for pinning down the specific version of the docker_registry.

**How this PR should be reviewed**

I've created the PR so it should be easy to test and review in a Vagrant box, things to note however are that I am passing `extra_args='-v'` as part of the `if docker_registry_version is defined` block.

The reason why I am doing this is because arguments supplied to `pip:` are stored in an array and when I start the `if` block an empty element is created and unless it is filled with something e.g. `extra_args='-v'` it will cause ansible to error

aalda commented:
Hi @actionjack 

Your proposal is welcomed, but I think you may get the same result in a cleaner way using the 'when' option. Something like this:

``` yaml
- name: Install latest version of Docker Registry
  pip: name=docker-registry state=present
  when: docker_registry_version is not defined

- name: Install a specific version of Docker Registry
  pip: name=docker-registry state=present version= {{ docker_registry_version }}
  when: docker_registry_version is defined      
```

actionjack commented:
@aalda Thanks for that I'll refactor test and update the PR in the morning UK time. T

Thanks again.

aalda commented:
Thanks to you :-)

actionjack committed:
Allow user to set the version of docker_registry

Let configuration manager pin the docker_registry to a specific version
that they want to be installed.

Thanks to @aalda for helping me refacter this into something cleaner and
easy to understand.

actionjack committed:
Update README.md

Update README.md to show usage of `docker_version_registry`

actionjack commented:
Hi @aalda I've refactored the PR
